src/createWagmiConfig.test.ts

```diff
@@ -0,0 +1,96 @@
+import { describe, expect, it, vi } from 'vitest';
+import { createConfig } from 'wagmi';
+import { http } from 'wagmi';
+import { base, baseSepolia } from 'wagmi/chains';
+import { coinbaseWallet } from 'wagmi/connectors';
+import { createWagmiConfig } from './createWagmiConfig';
+
+// Mock the imported modules
+vi.mock('wagmi', async () => {
+  const actual = await vi.importActual('wagmi');
+  return {
+    ...actual,
+    createConfig: vi.fn(),
+    createStorage: vi.fn(),
+  };
+});
+
+vi.mock('wagmi/chains', async () => {
+  const actual = await vi.importActual('wagmi/chains');
+  return {
+    ...actual,
+    base: { id: 8453 },
+    baseSepolia: { id: 84532 },
+  };
+});
+
+vi.mock('wagmi/connectors', async () => {
+  const actual = await vi.importActual('wagmi/connectors');
+  return {
+    ...actual,
+    coinbaseWallet: vi.fn(),
+  };
+});
+
+describe('createWagmiConfig', () => {
+  it('should create config with default values when no parameters are provided', () => {
+    createWagmiConfig({});
+    expect(createConfig).toHaveBeenCalledWith(
+      expect.objectContaining({
+        chains: [base, baseSepolia],
+        ssr: true,
+        transports: {
+          [base.id]: expect.any(Function),
+          [baseSepolia.id]: expect.any(Function),
+        },
+      }),
+    );
+    expect(coinbaseWallet).toHaveBeenCalledWith({
+      appName: undefined,
+      appLogoUrl: undefined,
+      preference: 'smartWalletOnly',
```

Comment:
Is this going to force the components to use smart wallet ? thinking this might be related to the issue i'm having where i'm unable to connect to EOA